src/device/device_ledger.cpp

```diff
@@ -303,7 +303,29 @@ namespace hw {
       offset += strlen(MONERO_VERSION);
       this->buffer_send[4] = offset-5;
       this->length_send = offset;
-      this->exchange();      
+      this->exchange();
+
+      ASSERT_X(this->length_recv>=3, "Communication error, less than tow three bytes received. Check your application version.");
+
+      unsigned int major = 0;
+      unsigned int minor = 0;
+      unsigned int micro = 0;
+      if (this->length_recv>=3) {
+        major = this->buffer_recv[0];
+        minor = this->buffer_recv[1];
+      }
+  
+      #define MINIMAL_VERSION_ERROR_MESSAGE \
+        "Unsuppotred device application version: "<<major<<"."<<minor<<"."<<micro<<"."\
+        " At least " << MINIMAL_MAJOR_APP_VERSION<<"."<<MINIMAL_MINOR_APP_VERSION<<"."<<MINIMAL_MICRO_APP_VERSION << " is requiered."
+      if (major < MINIMAL_MAJOR_APP_VERSION) {
+        ASSERT_X(false, MINIMAL_VERSION_ERROR_MESSAGE);
+      }
+      if (minor < MINIMAL_MINOR_APP_VERSION) {
+        ASSERT_X(false, MINIMAL_VERSION_ERROR_MESSAGE);
+      }
```

Comment:
That seems wrong. It will fail once you get to 2.0.0.
Maybe, eg:
```
 ASSERT_X(major*65536+minor*256+micro >= 0x010200, MINIMAL_VERSION_ERROR_MESSAGE);
```
(untested)
And you also get only one line with it :)

Reply:
you are right!. 
I will take your approach with shift: (major<<24) | (minor<<16) | micro